src/Async/index.js

```diff
@@ -322,6 +322,35 @@ function Async(fn) {
     }
   }
 
+  function bichain(l, r) {
+    if(!isFunction(l) || !isFunction(r)) {
+      throw new TypeError('Async.bichain: Functions required for both arguments')
+    }
+
+    return Async(function(rej, res) {
+      let cancel = unit
+      let innerCancel = unit
+
+      function setInnerCancel(mapFn) {
+        return function(x) {
+          const m = mapFn(x)
+
+          if(!isSameType(Async, m)) {
+            throw new TypeError(
+              'Async.bichain: Function must return another Async'
```

Comment:
We should reuse the same error message as we do not (and should not for security/log purposes) provide information on which function did not return an `Async`